projects/extension/src/background/SmoldotMediator.test.ts

```diff
@@ -0,0 +1,104 @@
+import { jest } from '@jest/globals';
+import { SmoldotClient} from 'smoldot';
+import {
+  customHealthResponder,
+  smoldotSpy,
+  respondWith
+} from '@substrate/smoldot-test-utils';
+import { SmoldotMediator } from './SmoldotMediator';
+import { JsonRpcRequest, JsonRpcResponse } from './types';
+import westend from '../../public/assets/westend.json';
+import { AppMediator } from './AppMediator';
+import { MockPort, MockConnectionManager } from '../mocks';
+
+let sc: SmoldotClient;
+let sm: SmoldotMediator;
+let appMed: AppMediator;
+let spyScSendJsonRpc: unknown;
+let spyScTerminate: unknown;
+let expectedResponse: number;
+let message:JsonRpcRequest;
+let idReturned: number;
+
+const createMessage = (id: number): JsonRpcRequest => ({
+  id: id,
+  jsonrpc: '2.0',
+  method: 'something',
+  params: []
+});
+
+beforeAll(async () => {
+  const healthResponses = [
```

Comment:
You don't need to use `customHealthResponder` and supply a sequence of healthy reponses when using `spoldotSpy`. If you omit the last argument it defaults to using an internal `healthyResponder` that takes care of responding with a healthy response each time it gets a "system_health" request.

Reply:
Even though I do not think that this invalidates the test - i have removed it as you suggested

Reply:
it doesn't but its a bit neater (less lines of code to distract from the essence of what we are testing - none of these tests have anything to do with health reponses).  Also I wasn't sure if you'd missed that it was possible to do this.